querier/src/cache_system/driver.rs

```diff
@@ -144,6 +192,21 @@ where
 /// - `Shared`: Allow the receiver to be cloned and be awaited from multiple places.
 type SharedReceiver<V> = Shared<BoxFuture<'static, Result<Arc<Mutex<V>>, Arc<RecvError>>>>;
 
+/// Running query
```

Comment:
How about something like:

```suggestion
/// State for coordinating the execution of a single running query.
```

or something?